src/pages/JobOwner/MyJobs/ClientJob/ClientJob.tsx

```diff
@@ -11,27 +11,49 @@ import { useGetOneJobPostQuery } from "services/jobPosts/setJobPostsAPI";
 import { useTranslation } from "react-i18next";
 import dayjs from "dayjs";
 import relativeTime from "dayjs/plugin/relativeTime";
-import { Button, Modal } from "antd";
+import { Button, Modal, notification } from "antd";
 import useModalHandler from "hooks/use-modal-handler";
+import defineContractStatus from "utils/functions/defineContractStatus";
 import { ContractStatusEnum } from "utils/enums";
+import { useCloseContractMutation } from "services/contract/contractApi";
 
 dayjs.extend(relativeTime);
 
 function ClientJob(): JSX.Element {
   const { jobId } = useParams();
   const { t } = useTranslation();
-  const { data } = useGetOneJobPostQuery({
+  const { data, refetch } = useGetOneJobPostQuery({
     id: Number(jobId),
   });
 
+  const [closeContract] = useCloseContractMutation();
+
   const { modal, toggleModal } = useModalHandler();
 
+  const onModalOk = async () => {
+    if (data?.contract) {
+      try {
+        await closeContract({
+          contractId: data?.contract.id,
+          endDate: new Date(),
+        });
+        toggleModal();
+        refetch();
+      } catch (e: any) {
```

Comment:
I believe it's better to use RTK query error feature with a useEffect, left a example below:

```typescript
const [closeContract, { isError, isLoading, isSuccess }] = useCloseContractMutation();

useEffect(() => {
    if (!isLoading && isError) {
      toast.error(t("ServerErrors.FETCH_ERROR"));
      return;
    }
    if (!isLoading && isSuccess) {
      toast.success(t("SearchWork.success"));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isLoading]);


const onModalOk = async () => {
  await closeContract({
    contractId: data?.contract.id,
    endDate: new Date(),
  });
  toggleModal();
  refetch();
}
```

Reply:
done